### PR TITLE
msglint: init at 1.04

### PR DESCRIPTION
**Pull request by apeyroux**
###### Motivation for this change

init msglint 

**! I don't understand the license the developer wanted to use !**

```
(C) Copyright 1991-1997,2002 by Chris Newman
All Rights Reserved.

Permission to use, copy, modify, distribute, and sell this software
and its documentation for any purpose is hereby granted without fee,
provided that the above copyright notice appear in all copies, and
that the name of Chris Newman not be used in advertising or publicity
pertaining to distribution of the software without specific, written
prior permission.  Chris Newman makes no representations about the
suitability of this software for any purpose.  It is provided "as is"
without express or implied warranty.

CHRIS NEWMAN DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO
EVENT SHALL CHRIS NEWMAN BE LIABLE FOR ANY SPECIAL, INDIRECT OR
CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF
USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/msglint/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    mkdir -p $out/bin
+    cp msglint $out/bin/msglint
+  '';
+
+  meta = with stdenv.lib; {
+    description  = "RFC 822/MIME/DSN/MDN/Tracking-Status message validator.";
+    homepage     = "https://tools.ietf.org/tools/msglint/";
+    downloadPage = "https://tools.ietf.org/tools/msglint/";
+    maintainers  = with maintainers; [ apeyroux ];
+    license      = licenses.unfree;
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:25]: The shipped `LICENSE` file looks like the [MIT license with an additional legal disclaimer](https://fedoraproject.org/wiki/Licensing:MIT#Old_Style_with_legal_disclaimer) for me.

```suggestion
    license      = licenses.mit;
```

File: pkgs/applications/misc/msglint/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp msglint $out/bin/msglint
+  '';
+
+  meta = with stdenv.lib; {
+    description  = "RFC 822/MIME/DSN/MDN/Tracking-Status message validator.";
+    homepage     = "https://tools.ietf.org/tools/msglint/";
+    downloadPage = "https://tools.ietf.org/tools/msglint/";
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:23]: `downloadPage` next to `homepage` with the same target is unnecessary imho.

File: pkgs/applications/misc/msglint/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    maintainers  = with maintainers; [ apeyroux ];
+    license      = licenses.unfree;
+    platforms    = [ "i686-linux" "x86_64-linux" "x86_64-darwin" ];
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:26]: On first sight, I do not see any platform limitations in the source code. Thus, I would omit the `platforms` entry.

File: pkgs/applications/misc/msglint/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation {
+  name = "msglint-1.04";
+
+  setSourceRoot = "sourceRoot=`pwd`";
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:6]: This is a very confusing way to set the `sourceRoot` to the current directory.

```suggestion
  sourceRoot = ".";
```

File: pkgs/applications/misc/msglint/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation {
+  name = "msglint-1.04";
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:4]: By convention, the package name and the version are separated in two variables. This might has the advantage for the `version` variable to be used within the `src` `url`.

```suggestion
  pname = "msglint";
  version = "1.04";
```

Next to this suggestion, you could add `rec` to the derivation and put the `${version}` within the source's `url`.

**Review by oxzi — CHANGES_REQUESTED** [review]
Thanks for this PR. The license looks like the MIT one for me.

**Comment by apeyroux** [comment]
@oxzi thx for your review

File: pkgs/applications/misc/msglint/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation {
+  pname = "msglint-1.04";
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:4]: If using both `pname` and `version`, the version number within the `pname` can be omitted.

```suggestion
  pname = "msglint";
```

File: pkgs/applications/misc/msglint/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  version = "1.04";
+
+  sourceRoot = ".";
+
+  src = fetchurl {
+    url = "https://tools.ietf.org/tools/msglint/msglint-1.04.tar.gz";
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:10]: You could use the `version` number from above within the URL. Therefore you must at first add `rec` between `mkDerivation` and the brace in line 3.

```suggestion
    url = "https://tools.ietf.org/tools/msglint/msglint-${version}.tar.gz";
```

**Review by oxzi — CHANGES_REQUESTED** [review]
Thanks for quickly adapting the proposed changes. Here are two more little things.

File: pkgs/applications/misc/msglint/default.nix (hunk 6 above)
Comment (apeyroux) [reply pkgs/applications/misc/msglint/default.nix]: I'm really stunned! :)

File: pkgs/applications/misc/msglint/default.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation {
```
Comment (oxzi) [inline pkgs/applications/misc/msglint/default.nix:3]: `rec` is missing here. Thus, the `version` variable is not available below.

```suggestion
stdenv.mkDerivation rec {
```

**Review by oxzi — CHANGES_REQUESTED** [review]

**Review by oxzi — APPROVED** [review]
LGTM

File: pkgs/applications/misc/msglint/default.nix [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  version = "1.04";
+
+  sourceRoot = ".";
+
+  src = fetchurl {
+    url = "https://tools.ietf.org/tools/msglint/msglint-${version}.tar.gz";
+    sha256 = "1yasj6nijip3z86m526bsssd5vgqrx233azvpz1c0z564wngvzbm";
+  };
+
+  patches = [ ./fixfree.patch ];
```
Comment (doronbehar) [inline pkgs/applications/misc/msglint/default.nix:14]:
```suggestion
  patches = [
    # What does this patch fix?
    ./fixfree.patch
  ];
```

File: pkgs/applications/misc/msglint/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp msglint $out/bin/msglint
```
Comment (doronbehar) [inline pkgs/applications/misc/msglint/default.nix:18]: I think this should work:

```suggestion
    runHook preInstall
    
    install msglint $out/bin/msglint
    
    runHook postInstall
```

File: pkgs/applications/misc/msglint/default.nix [hunk 11]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "msglint";
+  version = "1.04";
+
+  sourceRoot = ".";
```
Comment (doronbehar) [inline pkgs/applications/misc/msglint/default.nix:7]: Why is this needed?

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/msglint/default.nix (hunk 11 above)
Comment (oxzi) [reply pkgs/applications/misc/msglint/default.nix:7]: The tarball does not contain a directory, but extracts the files in the current working directory. Thus, one manually has to adjust the `sourceRoot`.

File: pkgs/applications/misc/msglint/default.nix (hunk 9 above)
Comment (oxzi) [reply pkgs/applications/misc/msglint/default.nix:14]: IIRC, this fixes potential segfaults for files passed as arguments. But yes, some describing comment would be good.

**Comment by SuperSandro2000** [comment]
@apeyroux ping

File: pkgs/applications/misc/msglint/default.nix (hunk 10 above)
Comment (apeyroux) [reply pkgs/applications/misc/msglint/default.nix:18]: should or can ? :)
